xdsl/backend/riscv/lowering/utils.py

```diff
@@ -33,7 +33,7 @@ def cast_to_regs(values: Iterable[SSAValue]) -> tuple[list[Operation], list[SSAV
         if not isinstance(value.type, riscv.RISCVRegisterType):
             register_type = register_type_for_type(value.type)
             cast_op = builtin.UnrealizedConversionCastOp.get(
-                (value,), (register_type(),)
+                (value,), (register_type(""),)
```

Comment:
Is the point of this PR not to remove `""` as a special register? Why can't the UNALLOCATED_INT register be reused?

Reply:
All register types with an empty name are the unallocated register for their class, and there's no API like `.unallocated_register` to use instead of this right now. I'm now thinking that that might be a good idea but I'd like to do that as a separate PR.

Reply:
Sounds good.